test/test_ncbi_taxa.py

```diff
@@ -386,284 +695,316 @@ def test_id_lookup_errors(fixture_ncbi_validator, test_input, expected_exception
+                (
+                    WARNING,
+                    "Non-canon usage: Cytophaga marina is synonym for "
+                    "Tenacibaculum maritimum",
+                ),
+                (INFO, "Match found for C marina"),
             ),
-            ("Escherichia coli", "species", 562, False, "strain", 561),
+            id="T maritimum from C marina",
         ),
-    ],
-)
-def test_taxa_search(fixture_ncbi_validator, test_input, expected):
-    """This test checks the results of searching for a specific taxon."""
-
-    fnd_tx = fixture_ncbi_validator.taxa_search(**test_input)
-
-    assert fnd_tx.name == expected[0]
-    assert fnd_tx.rank == expected[1]
-    assert fnd_tx.ncbi_id == expected[2]
-    assert fnd_tx.superseed == expected[3]
-
-    # Find last dictionary key
-    f_key = list(fnd_tx.taxa_hier.keys())[-1]
-
-    assert f_key == expected[1]
-
-    assert fnd_tx.taxa_hier[f_key] == (expected[0], expected[2], expected[5])
-    assert fnd_tx.orig == expected[4]
-
-
-# Now test that the search function logs errors correctly
-@pytest.mark.parametrize(
-    "test_input,expected_log_entries",
-    # Fine so empty
-    [
-        (
+        pytest.param(
             dict(
-                nnme="E coli",
-                taxah={"genus": "Escherichia", "species": "Escherichia coli"},
+                nnme="C marina",
+                taxon_hier=[("genus", "Cytophaga"), ("species", "Cytophaga marina")],
             ),
-            (),
-        ),
-        (
-            dict(nnme="Nonsense", taxah={"species": "Nonsense garbage"}),  # Nonsense
-            ((ERROR, "Taxa Nonsense cannot be found"),),
-        ),
-        (
-            dict(nnme="Morus", taxah={"genus": "Morus"}),  # ambiguous Morus
+            None,
+            pytest.raises(taxa.NCBIError),
+            None,
             (
                 (
-                    ERROR,
-                    "Taxa Morus cannot be found using only one taxonomic level,"
-                    " more should be provided",
+                    WARNING,
+                    "Non-canon usage: Cytophaga marina is synonym for "
+                    "Tenacibaculum maritimum",
                 ),
-            ),
-        ),
-        # Nonsense Morus
-        (
-            dict(nnme="N Morus", taxah={"family": "Nonsense", "genus": "Morus"}),
-            ((ERROR, "Provided parent taxa for N Morus not found"),),
-        ),
-        # Aegle Morus
-        (
-            dict(nnme="A Morus", taxah={"family": "Aegle", "genus": "Morus"}),
-            ((ERROR, "More than one possible parent taxa for A Morus found"),),
-        ),
-        # Eukaryote Morus
-        (
-            dict(nnme="E Morus", taxah={"superkingdom": "Eukaryota", "genus": "Morus"}),
-            (
                 (
                     ERROR,
-                    "Parent taxa for E Morus refers to multiple possible child taxa",
+                    "Taxonomy mismatch for Cytophaga at rank genus: expecting "
```

Comment:
I do worry about this one. Basically if someone is uploading old data and a species has been moved to a new genus, do they have to update the entire hierarchy? For kingdom -> family I guess not, but with genus you run into the problem that the species binomial gets checked against the genus name

Reply:
Though maybe this is handled by using timestamped database versions

Reply:
Yeah - the timestamping, I think. I don't see how we'd handle it otherwise.

Reply:
Yeah I suppose it's better here than in the GBIF case as is less a matter of opinion and more a matter of when they did their analysis, which is something time stamping allows us to account for